framework/src/base/ComputeFullJacobianThread.C

```diff
@@ -182,13 +182,23 @@ ComputeFullJacobianThread::computeInternalInterFaceJacobian(BoundaryID bnd_id)
       const std::vector<MooseSharedPointer<InterfaceKernel> > & int_ks = _interface_kernels.getActiveBoundaryObjects(bnd_id, _tid);
       for (const auto & interface_kernel : int_ks)
       {
+        if (!interface_kernel->isImplicit())
+          continue;
+
         unsigned int ivar = it.first->number();
-        if (interface_kernel->variable().number() == ivar && interface_kernel->isImplicit())
+        unsigned int jvar = it.second->number();
+
+        if (interface_kernel->variable().number() == ivar)
         {
-          unsigned int jvar = it.second->number();
-          interface_kernel->subProblem().prepareFaceShapes(jvar, _tid);
+          interface_kernel->subProblem().prepareFaceShapes(ivar, _tid);
           interface_kernel->subProblem().prepareNeighborShapes(jvar, _tid);
-          interface_kernel->computeOffDiagJacobian(jvar);
+          interface_kernel->computeElementOffDiagJacobian(jvar);
+        }
+        else if (interface_kernel->neighborVariable().number() == ivar)
```

Comment:
I think you missed my point.  If it's the same variable on both sides then you need to compute both `Element` AND `Neighbor` contributions here.  But with your current logic that's not possible.